File: docs/data-tools/codesnippet/CSharp/edit-data-in-datasets_7.cs
        private void FindErrors() 
        {
            if (dataSet1.HasErrors)
            {
                foreach (DataTable table in dataSet1.Tables)
                {
                    if (table.HasErrors)
                    {
                        foreach (DataRow row in table.Rows)
                        {
                            if (row.HasErrors)
                            {
                                // Process error here.
                            }
                        }
                    }
                }
            }
        }

File: docs/data-tools/codesnippet/VisualBasic/edit-data-in-datasets_7.vb
    Private Sub FindErrors()
        Dim table As Data.DataTable
        Dim row As Data.DataRow

        If DataSet1.HasErrors Then

            For Each table In DataSet1.Tables
                If table.HasErrors Then

                    For Each row In table.Rows
                        If row.HasErrors Then

                            ' Process error here.
                        End If
                    Next
                End If
            Next
        End If
    End Sub